contracts/WamplLoanRouter.sol

```diff
@@ -0,0 +1,111 @@
+pragma solidity ^0.8.3;
+
+import "./interfaces/ILoanRouter.sol";
+import "./interfaces/IBondController.sol";
+import "./interfaces/ITranche.sol";
+import "./interfaces/IButtonWrapper.sol";
+import "./interfaces/IWAMPL.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/IWamplLoanRouter.sol";
+import "./test/WAMPL.sol";
+
+/**
+ * @dev Wampl Loan Router built on top of a LoanRouter of your choosing
+ * to allow loans to be created with raw ampl instead of WAMPL
+ */
+contract WamplLoanRouter is IWamplLoanRouter {
+    ILoanRouter public immutable loanRouter;
+    IWAMPL public immutable wampl;
+
+    /**
+     * @dev Constructor for setting underlying loanRouter and wampl contracts
+     * @param _loanRouter The underlying loanRouter that does all the wrapping and trading
+     * @param _wampl The WAMPL contract to wrap AMPL in
+     */
+    constructor(ILoanRouter _loanRouter, IWAMPL _wampl) {
+        loanRouter = _loanRouter;
+        wampl = _wampl;
+    }
+
+    /**
+     * @inheritdoc IWamplLoanRouter
+     */
+    function wrapAndBorrow(
+        uint256 underlyingAmount,
+        IBondController bond,
+        IERC20 currency,
+        uint256[] memory sales,
+        uint256 minOutput
```

Comment:
If we keep it like this where we use an external loan router, we should still verify `minOutput` locally so we don't have to rely on external guarantees

Reply:
@marktoda Is this the check that you're referring to?
https://github.com/buttonwood-protocol/tranche/blob/main/contracts/LoanRouter.sol#L145-L146

Should I add this verification to WETH loan-router as well?